api/valor_api/schemas/metrics.py

```diff
@@ -418,7 +418,23 @@ def db_mapping(self, evaluation_id: int) -> dict:
         }
 
 
-class PrecisionRecallCurve(BaseModel):
+class _BasePrecisionRecallCurve(BaseModel):
+    """
+    Describes the parent class of our precision-recall curve metrics.
+
+    Attributes
+    ----------
+    label_key: str
+        The label key associated with the metric.
+    pr_curve_iou_threshold: float, optional
+        The IOU threshold to use when calculating precision-recall curves. Defaults to 0.5.
+    """
+
+    label_key: str
+    pr_curve_iou_threshold: float | None = None
```

Comment:
This structure is still confusing me as it could easily conform to the standard metrics format.

```
class Metric(BaseModel):
    """
    A metric response from the API.

    Attributes
    ----------
    type : str
        The type of metric.
    parameters : dict
        The parameters of the metric.
    value : float
        The value of the metric.
    label : Label
        The `Label` for the metric.
    """

    type: str
    parameters: dict | None = None
    value: float | dict | None = None
    label: Label | None = None
```

But instead pulls label key up and stores a dictionary by label value?

@ekorman What are your thoughts? I remember there being a reason for this organization early on.

Reply:
responded via slack